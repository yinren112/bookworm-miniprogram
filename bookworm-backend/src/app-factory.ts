// src/app-factory.ts
// Factory function to create Fastify app with proper database setup for tests

// Override the DATABASE_URL for tests using test containers
export const setupTestDatabase = () => {
  if (process.env.NODE_ENV === 'test' && process.env.TEST_CONTAINERS) {
    const containers = JSON.parse(process.env.TEST_CONTAINERS);
    const workerId = parseInt(process.env.VITEST_WORKER_ID || '1', 10);
    const databaseUrl = containers[workerId] || containers['1'];

    if (databaseUrl) {
      // Override the DATABASE_URL for this process
      process.env.DATABASE_URL = databaseUrl;
      console.log(`âœ… Test database URL set for worker ${workerId}: ${databaseUrl.substring(0, 30)}...`);
    }
  }
};

// Call this function before importing the main app to ensure the correct DATABASE_URL is used
export const createTestApp = async () => {
  setupTestDatabase();
  // Import after setting up the database URL
  const { buildApp } = await import('./index');
  return buildApp();
};