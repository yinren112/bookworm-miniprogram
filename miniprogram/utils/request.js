// miniprogram/utils/request.js - 统一的底层HTTP请求客户端
const config = require('../config');
const tokenUtil = require('./token');

/**
 * 生成简易的请求ID（用于追踪）
 * 格式：timestamp-random（如：1634567890123-a3f9）
 */
function generateRequestId() {
  const timestamp = Date.now();
  const random = Math.random().toString(36).substring(2, 6);
  return `${timestamp}-${random}`;
}

/**
 * 延迟函数
 * @param {number} ms - 延迟毫秒数
 */
const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));

/**
 * 判断是否应该重试
 * 只有 GET/HEAD 方法在网络错误或 5xx 错误时才重试
 * @param {string} method - HTTP 方法
 * @param {boolean} isNetworkError - 是否是网络错误
 * @param {number} statusCode - HTTP 状态码（如果有）
 */
function shouldRetry(method, isNetworkError, statusCode) {
  const isIdempotent = method === 'GET' || method === 'HEAD';
  if (!isIdempotent) return false;

  if (isNetworkError) return true;
  if (statusCode && statusCode >= 500 && statusCode < 600) return true;

  return false;
}

/**
 * 核心请求函数（内部递归调用）
 * @param {Object} options - 请求配置
 * @param {string} options.url - 请求路径（相对路径，会自动拼接 baseURL）
 * @param {string} options.method - HTTP 方法（默认 GET）
 * @param {Object} options.data - 请求数据
 * @param {boolean} options.requireAuth - 是否需要 Authorization header（默认 false）
 * @param {number} options.timeout - 超时时间（毫秒，默认 8000）
 * @param {number} attempt - 当前重试次数（内部使用）
 * @returns {Promise<any>} - 返回响应数据
 */
function performRequest(options, attempt = 0) {
  const {
    url,
    method = 'GET',
    data = {},
    requireAuth = false,
    timeout = 8000,
  } = options;

  // 指数退避延迟：100ms, 300ms, 900ms
  const RETRY_DELAYS = [100, 300, 900];
  const MAX_RETRIES = RETRY_DELAYS.length;

  const requestId = generateRequestId();
  const token = requireAuth ? tokenUtil.getToken() : null;

  return new Promise((resolve, reject) => {
    const headers = {
      'Content-Type': 'application/json',
      'X-Request-ID': requestId,
    };

    if (requireAuth && token) {
      headers.Authorization = `Bearer ${token}`;
    }

    // eslint-disable-next-line no-restricted-syntax -- utils/request.js 是唯一允许直接调用 wx.request 的文件
    wx.request({
      url: `${config.apiBaseUrl}${url}`,
      method,
      data,
      timeout,
      header: headers,
      success: async (res) => {
        // 成功响应（2xx）
        if (res.statusCode >= 200 && res.statusCode < 300) {
          resolve(res.data);
          return;
        }

        // 检查是否应该重试（只对 GET/HEAD 且是 5xx 错误）
        if (shouldRetry(method, false, res.statusCode) && attempt < MAX_RETRIES) {
          const delayMs = RETRY_DELAYS[attempt];
          await sleep(delayMs);
          try {
            const retryResult = await performRequest(options, attempt + 1);
            resolve(retryResult);
          } catch (retryError) {
            reject(retryError);
          }
          return;
        }

        // 非重试情况，返回错误
        const errorPayload = res.data && typeof res.data === 'object'
          ? { ...res.data, statusCode: res.statusCode, requestId }
          : {
              message: `请求失败 (${res.statusCode})`,
              statusCode: res.statusCode,
              requestId
            };
        reject(errorPayload);
      },
      fail: async (error) => {
        // 网络错误，检查是否应该重试
        if (shouldRetry(method, true, null) && attempt < MAX_RETRIES) {
          const delayMs = RETRY_DELAYS[attempt];
          await sleep(delayMs);
          try {
            const retryResult = await performRequest(options, attempt + 1);
            resolve(retryResult);
          } catch (retryError) {
            reject(retryError);
          }
          return;
        }

        // 非重试情况或重试次数用尽，返回网络错误
        reject({
          message: '网络请求失败',
          errorCode: 'NETWORK_ERROR',
          requestId,
          detail: error,
        });
      },
    });
  });
}

/**
 * 统一的请求客户端
 * @param {Object} options - 请求配置
 * @param {string} options.url - 请求路径（相对路径）
 * @param {string} options.method - HTTP 方法（默认 GET）
 * @param {Object} options.data - 请求数据
 * @param {boolean} options.requireAuth - 是否需要 Authorization header（默认 false）
 * @param {number} options.timeout - 超时时间（毫秒，默认 8000）
 * @returns {Promise<any>} - 返回响应数据
 */
function request(options) {
  return performRequest(options, 0);
}

module.exports = {
  request,
};
