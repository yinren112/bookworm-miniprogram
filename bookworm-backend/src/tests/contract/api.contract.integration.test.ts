/**
 * API Contract Tests
 *
 * 目标：冻结公共 API 的 JSON 响应结构，防止意外的破坏性变更
 *
 * 策略：
 * - 黑箱测试：仅验证 HTTP 响应的 JSON 结构
 * - 快照测试：使用 Jest snapshots 冻结响应格式
 * - Normalize：去除动态字段（timestamp, token, ID等）
 *
 * 重要：
 * - CI 中禁止自动更新快照 (--updateSnapshot)
 * - 快照变更需要 Code Review 并附上破坏性/兼容性说明
 */

import { describe, it, expect, beforeAll, afterAll, beforeEach } from "vitest";
import request from "supertest";
import { createTestApp } from "../../app-factory";
import { FastifyInstance } from "fastify";
import { getPrismaClientForWorker, createTestUser, createTestInventoryItems } from '../globalSetup';
import { PrismaClient } from "@prisma/client";

/**
 * 规范化函数：去除动态字段
 *
 * 动态字段包括：
 * - 时间戳: createdAt, updatedAt, paymentExpiresAt, paid_at, cancelled_at, etc.
 * - 令牌: token, access_token, refresh_token
 * - 动态ID: 某些情况下的 ID（如果测试之间不稳定）
 * - 签名: signature, nonce
 * - 取货码: pickupCode（每次生成不同）
 */
function normalize(input: any): any {
  if (input == null) return input;

  if (Array.isArray(input)) return input.map(normalize);

  if (typeof input === 'object') {
    const out: any = {};
    for (const [key, value] of Object.entries(input)) {
      const lowerKey = key.toLowerCase();

      // 动态 ID/令牌/签名类
      if (/(^|_)(id|token|nonce|signature|sign)$/.test(lowerKey) && typeof value === 'string') {
        out[key] = '[DYNAMIC]';
        continue;
      }

      // 时间戳类
      if (/(^|_)(created|updated|paid|refunded)at$/.test(lowerKey)) {
        out[key] = '[DYNAMIC_TIMESTAMP]';
        continue;
      }

      // 业务敏感动态字段
      if (lowerKey === 'pickupcode' || lowerKey === 'pickup_code') { out[key] = '[DYNAMIC_PICKUP_CODE]'; continue; }
      if (lowerKey === 'openid') { out[key] = '[DYNAMIC_OPENID]'; continue; }
      if (lowerKey === 'phonenumber' || lowerKey === 'phone' || lowerKey === 'phone_number') { out[key] = '[DYNAMIC_PHONE]'; continue; }

      // slug：包含时间戳/递增后缀的场景，如 test-content-1697... 或 ...-123
      if (lowerKey === 'slug' && typeof value === 'string') {
        if (/-\d+($|[^a-zA-Z])/.test(value) || /\d{6,}/.test(value)) {
          out[key] = '[DYNAMIC_SLUG]';
          continue;
        }
      }

      out[key] = normalize(value);
    }
    return out;
  }

  // ISO 时间字符串兜底
  if (typeof input === 'string' && /^\d{4}-\d{2}-\d{2}T/.test(input)) return '[DYNAMIC_TIMESTAMP]';
  return input;
}

describe("API Contract Tests", () => {
  let app: FastifyInstance;
  let prisma: PrismaClient;
  let staffToken: string;
  let userToken: string;
  let testUserId: number;
  let staffUserId: number;

  beforeAll(async () => {
    prisma = getPrismaClientForWorker();
    app = await createTestApp();
    await app.ready();
  });

  afterAll(async () => {
    await app.close();
  });

  beforeEach(async () => {
    const staffUser = await createTestUser("STAFF");
    const regularUser = await createTestUser("USER");

    staffToken = staffUser.token;
    userToken = regularUser.token;
    testUserId = regularUser.userId;
    staffUserId = staffUser.userId;
  });

  describe("Authentication APIs", () => {
    it("POST /api/auth/login - 登录成功返回token和userId", async () => {
      // 测试目标：验证登录响应结构
      const mockOpenid = `test-openid-${Date.now()}`;

      // 创建mock用户
      await prisma.user.create({
        data: {
          openid: mockOpenid,
          role: "USER",
          status: "REGISTERED",
        }
      });

      const res = await request(app.server)
        .post("/api/auth/login")
        .send({ code: `test-code-${Date.now()}` });

      expect(res.status).toBe(200);
      expect(normalize(res.body)).toMatchSnapshot();
    });
  });

  describe("Inventory APIs", () => {
    it("GET /api/inventory/available - 获取可用库存列表", async () => {
      // 前置：创建一些库存
      await createTestInventoryItems(3);

      const res = await request(app.server)
        .get("/api/inventory/available")
        .query({ page: 1, limit: 10 });

      expect(res.status).toBe(200);

      // 验证基本结构
      expect(res.body).toHaveProperty('data');
      expect(Array.isArray(res.body.data)).toBe(true);

      // 快照测试（去除动态字段和ID）
      const normalizedBody = {
        ...normalize(res.body),
        // 进一步规范化：去除 ID，因为它们在不同测试运行中可能不同
        data: res.body.data.map((item: any) => ({
          ...normalize(item),
          id: '[DYNAMIC_ID]',
          booksku: {
            ...normalize(item.booksku),
            id: '[DYNAMIC_ID]',
            bookmaster: {
              ...normalize(item.booksku?.bookmaster),
              id: '[DYNAMIC_ID]'
            }
          }
        }))
      };

      expect(normalizedBody).toMatchSnapshot();
    });

    it("GET /api/inventory/item/:id - 获取单个库存详情", async () => {
      const [itemId] = await createTestInventoryItems(1);

      const res = await request(app.server)
        .get(`/api/inventory/item/${itemId}`);

      expect(res.status).toBe(200);

      // 规范化ID
      const normalizedBody = {
        ...normalize(res.body),
        id: '[DYNAMIC_ID]',
        sku_id: '[DYNAMIC_ID]',
        booksku: res.body.booksku ? {
          ...normalize(res.body.booksku),
          id: '[DYNAMIC_ID]',
          master_id: '[DYNAMIC_ID]',
          bookmaster: res.body.booksku.bookmaster ? {
            ...normalize(res.body.booksku.bookmaster),
            id: '[DYNAMIC_ID]'
          } : undefined
        } : undefined
      };

      expect(normalizedBody).toMatchSnapshot();
    });
  });

  describe("Order APIs", () => {
    it("POST /api/orders/create - 创建订单", async () => {
      const inventoryItemIds = await createTestInventoryItems(2);

      const res = await request(app.server)
        .post("/api/orders/create")
        .set("Authorization", `Bearer ${userToken}`)
        .send({ inventoryItemIds });

      expect(res.status).toBe(201);

      // 规范化
      const normalizedBody = {
        ...normalize(res.body),
        id: '[DYNAMIC_ID]',
        userId: '[DYNAMIC_USER_ID]'
      };

      expect(normalizedBody).toMatchSnapshot();
    });

    it("GET /api/orders/:id - 获取订单详情", async () => {
      // 前置：创建一个订单
      const inventoryItemIds = await createTestInventoryItems(1);
      const createRes = await request(app.server)
        .post("/api/orders/create")
        .set("Authorization", `Bearer ${userToken}`)
        .send({ inventoryItemIds });

      const orderId = createRes.body.id;

      const res = await request(app.server)
        .get(`/api/orders/${orderId}`)
        .set("Authorization", `Bearer ${userToken}`);

      expect(res.status).toBe(200);

      // 规范化
      const normalizedBody = {
        ...normalize(res.body),
        id: '[DYNAMIC_ID]',
        user_id: '[DYNAMIC_USER_ID]',
        items: res.body.items?.map((item: any) => ({
          ...normalize(item),
          id: '[DYNAMIC_ID]',
          order_id: '[DYNAMIC_ORDER_ID]',
          inventory_item_id: '[DYNAMIC_ITEM_ID]',
          inventory_item: item.inventory_item ? {
            ...normalize(item.inventory_item),
            id: '[DYNAMIC_ID]',
            sku_id: '[DYNAMIC_SKU_ID]'
          } : undefined
        }))
      };

      expect(normalizedBody).toMatchSnapshot();
    });

    it("GET /api/orders/my - 获取用户订单列表", async () => {
      // 前置：创建几个订单
      const inventoryItemIds1 = await createTestInventoryItems(1);
      const inventoryItemIds2 = await createTestInventoryItems(1);

      await request(app.server)
        .post("/api/orders/create")
        .set("Authorization", `Bearer ${userToken}`)
        .send({ inventoryItemIds: inventoryItemIds1 });

      await request(app.server)
        .post("/api/orders/create")
        .set("Authorization", `Bearer ${userToken}`)
        .send({ inventoryItemIds: inventoryItemIds2 });

      const res = await request(app.server)
        .get("/api/orders/my")
        .set("Authorization", `Bearer ${userToken}`)
        .query({ limit: 10 });

      expect(res.status).toBe(200);

      // 验证基本结构
      expect(res.body).toHaveProperty('data');
      expect(Array.isArray(res.body.data)).toBe(true);
      expect(res.body).toHaveProperty('meta');

      // 规范化
      const normalizedBody = {
        data: res.body.data.map((order: any) => ({
          ...normalize(order),
          id: '[DYNAMIC_ID]',
          user_id: '[DYNAMIC_USER_ID]'
        })),
        meta: {
          nextCursor: res.body.meta.nextCursor ? '[DYNAMIC_CURSOR]' : null
        }
      };

      expect(normalizedBody).toMatchSnapshot();
    });
  });

  describe("Books Recommendation APIs", () => {
    it("GET /api/books/recommendations - 获取推荐书籍（无用户资料）", async () => {
      const res = await request(app.server)
        .get("/api/books/recommendations")
        .set("Authorization", `Bearer ${userToken}`);

      expect(res.status).toBe(200);

      // 快照测试
      expect(normalize(res.body)).toMatchSnapshot();
    });

    it("GET /api/books/recommendations - 获取推荐书籍（有用户资料）", async () => {
      // 前置：创建用户资料
      await prisma.userProfile.create({
        data: {
          user_id: testUserId,
          enrollment_year: 2023,
          major: "计算机科学",
          class_name: "1班"
        }
      });

      const res = await request(app.server)
        .get("/api/books/recommendations")
        .set("Authorization", `Bearer ${userToken}`);

      expect(res.status).toBe(200);

      // 快照测试
      const normalizedBody = {
        ...normalize(res.body),
        items: res.body.items?.map((item: any) => ({
          ...normalize(item),
          id: '[DYNAMIC_ID]',
          list_id: '[DYNAMIC_LIST_ID]',
          sku_id: '[DYNAMIC_SKU_ID]'
        }))
      };

      expect(normalizedBody).toMatchSnapshot();
    });
  });

  describe("Content APIs", () => {
    it("GET /api/content/:slug - 获取静态内容", async () => {
      // 前置：创建一个内容
      const testSlug = `test-content-${Date.now()}`;
      await prisma.content.create({
        data: {
          slug: testSlug,
          title: "测试内容",
          body: "这是测试内容的正文",
        }
      });

      const res = await request(app.server)
        .get(`/api/content/${testSlug}`);

      expect(res.status).toBe(200);

      // 快照测试
      const normalizedBody = {
        ...normalize(res.body),
        id: '[DYNAMIC_ID]'
      };

      expect(normalizedBody).toMatchSnapshot();
    });
  });

  describe("User APIs", () => {
    it("GET /api/users/me - 获取当前用户信息", async () => {
      const res = await request(app.server)
        .get("/api/users/me")
        .set("Authorization", `Bearer ${userToken}`);

      expect(res.status).toBe(200);

      // 快照测试
      const normalizedBody = {
        ...normalize(res.body),
        id: '[DYNAMIC_ID]'
      };

      expect(normalizedBody).toMatchSnapshot();
    });
  });

  describe("Error Response Contracts", () => {
    it("401 Unauthorized - 缺少认证令牌", async () => {
      const res = await request(app.server)
        .get("/api/orders/my");

      expect(res.status).toBe(401);
      expect(normalize(res.body)).toMatchSnapshot();
    });

    it("403 Forbidden - 权限不足", async () => {
      const res = await request(app.server)
        .get("/api/orders/pending-pickup")
        .set("Authorization", `Bearer ${userToken}`); // 普通用户访问员工接口

      expect(res.status).toBe(403);
      expect(normalize(res.body)).toMatchSnapshot();
    });

    it("404 Not Found - 资源不存在", async () => {
      const res = await request(app.server)
        .get("/api/orders/99999")
        .set("Authorization", `Bearer ${userToken}`);

      expect(res.status).toBe(404);
      expect(normalize(res.body)).toMatchSnapshot();
    });

    it("400 Bad Request - 验证错误", async () => {
      const res = await request(app.server)
        .post("/api/orders/create")
        .set("Authorization", `Bearer ${userToken}`)
        .send({ inventoryItemIds: "not-an-array" }); // 类型错误

      expect(res.status).toBe(400);
      expect(normalize(res.body)).toMatchSnapshot();
    });

    it("409 Conflict - 业务冲突", async () => {
      // 前置：创建一个订单（PENDING_PAYMENT）
      const inventoryItemIds1 = await createTestInventoryItems(1);
      await request(app.server)
        .post("/api/orders/create")
        .set("Authorization", `Bearer ${userToken}`)
        .send({ inventoryItemIds: inventoryItemIds1 });

      // 尝试创建第二个订单（应该冲突）
      const inventoryItemIds2 = await createTestInventoryItems(1);
      const res = await request(app.server)
        .post("/api/orders/create")
        .set("Authorization", `Bearer ${userToken}`)
        .send({ inventoryItemIds: inventoryItemIds2 });

      expect(res.status).toBe(409);
      expect(normalize(res.body)).toMatchSnapshot();
    });
  });
});
