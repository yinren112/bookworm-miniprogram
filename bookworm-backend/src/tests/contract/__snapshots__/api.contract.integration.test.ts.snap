// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`API Contract Tests > Authentication APIs > POST /api/auth/login - 登录成功返回token和userId 1`] = `
{
  "merged": false,
  "token": "[DYNAMIC]",
  "userId": 4,
}
`;

exports[`API Contract Tests > Books Recommendation APIs > GET /api/books/recommendations - 获取推荐书籍（无用户资料） 1`] = `
{
  "count": 0,
  "recommendations": [],
}
`;

exports[`API Contract Tests > Books Recommendation APIs > GET /api/books/recommendations - 获取推荐书籍（有用户资料） 1`] = `
{
  "count": 0,
  "items": undefined,
  "recommendations": [],
}
`;

exports[`API Contract Tests > Content APIs > GET /api/content/:slug - 获取静态内容 1`] = `
{
  "body": "这是测试内容的正文",
  "createdAt": "[DYNAMIC_TIMESTAMP]",
  "id": "[DYNAMIC_ID]",
  "slug": "[DYNAMIC_SLUG]",
  "title": "测试内容",
  "updatedAt": "[DYNAMIC_TIMESTAMP]",
}
`;

exports[`API Contract Tests > Error Response Contracts > 400 Bad Request - 验证错误 1`] = `
{
  "code": "VALIDATION_ERROR",
  "details": [
    {
      "instancePath": "/inventoryItemIds",
      "keyword": "type",
      "message": "must be array",
      "params": {
        "type": "array",
      },
      "schemaPath": "#/properties/inventoryItemIds/type",
    },
  ],
  "message": "Request validation failed",
}
`;

exports[`API Contract Tests > Error Response Contracts > 401 Unauthorized - 缺少认证令牌 1`] = `
{
  "code": "UNAUTHORIZED",
  "message": "Missing authorization header",
}
`;

exports[`API Contract Tests > Error Response Contracts > 403 Forbidden - 权限不足 1`] = `
{
  "code": "FORBIDDEN",
  "message": "Forbidden",
}
`;

exports[`API Contract Tests > Error Response Contracts > 404 Not Found - 资源不存在 1`] = `
{
  "code": "ORDER_NOT_FOUND",
  "message": "Order not found",
}
`;

exports[`API Contract Tests > Error Response Contracts > 409 Conflict - 业务冲突 1`] = `
{
  "code": "CONCURRENT_PENDING_ORDER",
  "message": "您有一个正在付款的订单，请先完成付款或等待订单过期",
}
`;

exports[`API Contract Tests > Inventory APIs > GET /api/inventory/available - 获取可用库存列表 1`] = `
{
  "data": [
    {
      "booksku": {
        "bookmaster": {
          "author": "Test Author",
          "id": "[DYNAMIC_ID]",
          "isbn13": "9781000000001",
          "original_price": 10000,
          "publisher": "Test Publisher",
          "title": "Test Book 1-1",
        },
        "cover_image_url": "https://example.com/cover.jpg",
        "edition": "1st Edition",
        "id": "[DYNAMIC_ID]",
      },
      "condition": "GOOD",
      "id": "[DYNAMIC_ID]",
      "selling_price": 8000,
      "status": "in_stock",
    },
    {
      "booksku": {
        "bookmaster": {
          "author": "Test Author",
          "id": "[DYNAMIC_ID]",
          "isbn13": "9781000000001",
          "original_price": 10000,
          "publisher": "Test Publisher",
          "title": "Test Book 1-1",
        },
        "cover_image_url": "https://example.com/cover.jpg",
        "edition": "1st Edition",
        "id": "[DYNAMIC_ID]",
      },
      "condition": "GOOD",
      "id": "[DYNAMIC_ID]",
      "selling_price": 8000,
      "status": "in_stock",
    },
    {
      "booksku": {
        "bookmaster": {
          "author": "Test Author",
          "id": "[DYNAMIC_ID]",
          "isbn13": "9781000000001",
          "original_price": 10000,
          "publisher": "Test Publisher",
          "title": "Test Book 1-1",
        },
        "cover_image_url": "https://example.com/cover.jpg",
        "edition": "1st Edition",
        "id": "[DYNAMIC_ID]",
      },
      "condition": "GOOD",
      "id": "[DYNAMIC_ID]",
      "selling_price": 8000,
      "status": "in_stock",
    },
  ],
  "meta": {
    "currentPage": 1,
    "itemsPerPage": 10,
    "totalItems": 3,
    "totalPages": 1,
  },
}
`;

exports[`API Contract Tests > Inventory APIs > GET /api/inventory/item/:id - 获取单个库存详情 1`] = `
{
  "bookSku": {
    "bookMaster": {
      "author": "Test Author",
      "id": 1,
      "isbn13": "9781000000002",
      "original_price": 10000,
      "publisher": "Test Publisher",
      "title": "Test Book 1-2",
    },
    "cover_image_url": "https://example.com/cover.jpg",
    "edition": "1st Edition",
    "id": 1,
  },
  "booksku": undefined,
  "condition": "GOOD",
  "id": "[DYNAMIC_ID]",
  "selling_price": 8000,
  "sku_id": "[DYNAMIC_ID]",
  "status": "in_stock",
}
`;

exports[`API Contract Tests > Order APIs > GET /api/orders/:id - 获取订单详情 1`] = `
{
  "cancelled_at": null,
  "createdAt": "[DYNAMIC_TIMESTAMP]",
  "id": "[DYNAMIC_ID]",
  "items": undefined,
  "orderItem": [
    {
      "id": 1,
      "inventoryItem": {
        "bookSku": {
          "bookMaster": {
            "author": "Test Author",
            "id": 1,
            "isbn13": "9781000000004",
            "original_price": 10000,
            "publisher": "Test Publisher",
            "title": "Test Book 1-4",
          },
          "cover_image_url": "https://example.com/cover.jpg",
          "edition": "1st Edition",
          "id": 1,
        },
        "condition": "GOOD",
        "id": 1,
        "selling_price": 8000,
      },
      "inventory_item_id": 1,
      "order_id": 1,
    },
  ],
  "paid_at": null,
  "paymentExpiresAt": "[DYNAMIC_TIMESTAMP]",
  "pickup_code": "[DYNAMIC_PICKUP_CODE]",
  "status": "PENDING_PAYMENT",
  "total_amount": "80",
  "user_id": "[DYNAMIC_USER_ID]",
}
`;

exports[`API Contract Tests > Order APIs > GET /api/orders/my - 获取用户订单列表 1`] = `
{
  "data": [
    {
      "cancelled_at": null,
      "createdAt": "[DYNAMIC_TIMESTAMP]",
      "id": "[DYNAMIC_ID]",
      "orderItem": [
        {
          "id": 1,
          "inventoryItem": {
            "bookSku": {
              "bookMaster": {
                "author": "Test Author",
                "id": 1,
                "isbn13": "9781000000005",
                "original_price": 10000,
                "publisher": "Test Publisher",
                "title": "Test Book 1-5",
              },
              "cover_image_url": "https://example.com/cover.jpg",
              "edition": "1st Edition",
              "id": 1,
            },
            "condition": "GOOD",
            "id": 1,
            "selling_price": 8000,
          },
          "inventory_item_id": 1,
          "order_id": 1,
        },
      ],
      "paid_at": null,
      "paymentExpiresAt": "[DYNAMIC_TIMESTAMP]",
      "pickup_code": "[DYNAMIC_PICKUP_CODE]",
      "status": "PENDING_PAYMENT",
      "total_amount": "80",
      "user_id": "[DYNAMIC_USER_ID]",
    },
  ],
  "meta": {
    "nextCursor": null,
  },
}
`;

exports[`API Contract Tests > Order APIs > POST /api/orders/create - 创建订单 1`] = `
{
  "cancelled_at": null,
  "completed_at": null,
  "createdAt": "[DYNAMIC_TIMESTAMP]",
  "id": "[DYNAMIC_ID]",
  "notes": null,
  "paid_at": null,
  "paymentExpiresAt": "[DYNAMIC_TIMESTAMP]",
  "pickupExpiresAt": null,
  "pickup_code": "[DYNAMIC_PICKUP_CODE]",
  "settlementType": null,
  "status": "PENDING_PAYMENT",
  "totalWeightKg": null,
  "total_amount": "160",
  "type": "PURCHASE",
  "unitPrice": null,
  "userId": "[DYNAMIC_USER_ID]",
  "user_id": 2,
  "voucherFaceValue": null,
}
`;

exports[`API Contract Tests > User APIs > GET /api/users/me - 获取当前用户信息 1`] = `
{
  "createdAt": "[DYNAMIC_TIMESTAMP]",
  "id": "[DYNAMIC_ID]",
  "phone_number": "[DYNAMIC_PHONE]",
  "role": "USER",
}
`;
